client/electron/go_vpn_tunnel.ts

```diff
@@ -310,17 +319,39 @@ class GoTun2socks {
   }
 
   /**
-   * Checks connectivity and exits with the string of stdout.
+   * Checks connectivity of the server specified in `config`.
+   * Checks whether proxy server is reachable, whether the network and proxy support UDP forwarding
+   * and validates the proxy credentials.
    *
+   * @returns A boolean indicating whether UDP forwarding is supported.
    * @throws ProcessTerminatedExitCodeError if tun2socks failed to run successfully.
    */
-  checkConnectivity() {
+  async checkConnectivity(config: TransportConfigJson): Promise<boolean> {
```

Comment:
Can you move this to a standalone function?

There's no real dependency on internal state, only `this.process`, which you can replace with `new ChildProcessHelper(pathToEmbeddedTun2socksBinary())`.

Reply:
Done